File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character> map1=new HashMap<>();
        HashMap<Character,Character> map2=new HashMap<>();
        for(int index=0;index<s.length();index++){
            if(map1.containsKey(s.charAt(index))){
                if(map1.get(s.charAt(index))!=t.charAt(index)){
                    return false;
                }
            }
            else{
                map1.put(s.charAt(index),t.charAt(index));
            }

            if(map2.containsKey(t.charAt(index))){
                if(map2.get(t.charAt(index))!=s.charAt(index)){
                    return false;
                }
            }
            else{
                map2.put(t.charAt(index),s.charAt(index));
            }
        }
        return true;
    }
}

File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        
        d1={}
        d2={}
        for i in range(len(s)):
            ch1=s[i]
            ch2=t[i]
            if(ch1 in d1 and d1[ch1]!=ch2):
                return False
            if(ch2 in d2 and d2[ch2]!=ch1):
                return False
            d1[ch1]=ch2
            d2[ch2]=ch1
        return True
                
